File: dynamo/python/delete/dynamo_delete.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	dynamo_delete.py
#
#					Oct/21/2017
# --------------------------------------------------------------------
import sys
import boto3

# --------------------------------------------------------------------
sys.stderr.write("*** 開始 ***\n")
key_in = sys.argv[1]
print("%s" % key_in)

dynamodb = boto3.resource('dynamodb',endpoint_url="http://atami:8000")

table = dynamodb.Table('cities')

response = table.delete_item(
	Key={
            'key': key_in,
        }
)
#
sys.stderr.write("*** 終了 ***\n")
# --------------------------------------------------------------------

File: dynamo/node/delete/dynamo_delete.js
#! /usr/bin/node
// ---------------------------------------------------------------
//	dynamo_delete.js
//
//					Dec/19/2017
//
// ---------------------------------------------------------------
function delete_proc (docClient,key_in)
{
	const tbl_name = "cities"

	var params = {
		TableName: tbl_name,
		Key: {"key": {"S": key_in} }
			}

	docClient.delete(params, function(err, data)
		{
		if (err) {
			console.error("Unable to get cities",JSON.stringify(err, null, 2))
		
			console.error("key = " + key)
			}
		 else {
			}
		})

}

// ---------------------------------------------------------------
console.error ("*** 開始 ***")
const key_in=process.argv[2]

console.error ("key_in = " + key_in)

var AWS = require("aws-sdk")

AWS.config.update({
	region: "ap-northeast-1",
	endpoint: "http://localhost:8000"
})

var docClient = new AWS.DynamoDB.DocumentClient();

delete_proc (docClient,key_in)

console.error ("*** 終了 ***")

// ---------------------------------------------------------------
